react/src/components/ComputeSessionNodeItems/TerminateSessionModal.tsx

```diff
@@ -0,0 +1,381 @@
+      @relay(plural: true) {
+        id
+        row_id
+        name
+        scaling_group @required(action: NONE)
+        kernel_nodes(first: 100) {
+          edges {
+            node {
+              container_id
+              agent_id
+            }
+          }
+        }
+      }
+    `,
+    sessionFrgmt,
+  );
+  const [isForce, setIsForce] = useState(false);
+  const userRole = useCurrentUserRole();
+
+  const baiClient = useSuspendedBackendaiClient();
+
+  const currentProject = useCurrentProjectValue();
+
+  const terminateMutation = useTanMutation({
+    mutationFn: async (session: Session) => {
+      return terminateApp(
+        session,
+        baiClient._config.accessKey,
+        currentProject.id,
+        baiClient,
+      )
+        .catch((e) => {
+          return {
+            error: e,
+          };
+        })
+        .then((result) => {
+          const err = result?.error;
+          if (
+            err === undefined || //no error
+            (err && // Even if wsproxy address is invalid, session must be deleted.
+              err.message &&
+              (err.statusCode === 404 || err.statusCode === 500))
+          ) {
+            // BAI client destroy try to request 3times as default
+            return baiClient.destroy(
+              session.row_id,
+              baiClient._config.accessKey,
+              isForce,
+            );
+          } else {
+            throw err;
+          }
+        });
+    },
+  });
+  const relayEvn = useRelayEnvironment();
+  const painKiller = usePainKiller();
+  const { upsertNotification } = useSetBAINotification();
+
+  return (
+    <BAIModal
+      centered
+      title={t('session.TerminateSession')}
+      open={openTerminateModal}
+      confirmLoading={terminateMutation.isPending}
+      onOk={() => {
+        if (sessions[0]?.row_id) {
+          const session = sessions[0];
+          terminateMutation
+            .mutateAsync(session)
+            .then(() => {
+              setIsForce(false);
+              onRequestClose(true);
+            })
+            .catch((err) => {
+              // this.notification.text = PainKiller.relieve(err.title);
+              // this.notification.detail = err.message;
+              // this.notification.show(true, err);
+              upsertNotification({
+                message: painKiller.relieve(err?.title),
+                description: err?.message,
+                open: true,
+              });
+            })
+            .finally(() => {
+              // TODO: remove below code after session list migration to React
+              const event = new CustomEvent(
+                'backend-ai-session-list-refreshed',
+                {
+                  detail: 'running',
+                },
+              );
+              document.dispatchEvent(event);
+
+              // refetch session node
+              return fetchQuery<TerminateSessionModalRefetchQuery>(
```

Comment:
Not related to this PR, but it would be nice if the session in the query used for `EndpointDetailPage` was also changed to session node instead of UUID.